akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala

```diff
@@ -32,58 +32,59 @@ private[akka] object PersistentBehaviorImpl {
 
 @InternalApi
 private[akka] final case class PersistentBehaviorImpl[Command, Event, State](
-  persistenceId:     String,
-  emptyState:        State,
-  commandHandler:    PersistentBehaviors.CommandHandler[Command, Event, State],
-  eventHandler:      PersistentBehaviors.EventHandler[State, Event],
-  journalPluginId:   Option[String]                                              = None,
-  snapshotPluginId:  Option[String]                                              = None,
-  recoveryCompleted: (ActorContext[Command], State) ⇒ Unit                       = ConstantFun.scalaAnyTwoToUnit,
-  tagger:            Event ⇒ Set[String]                                         = (_: Event) ⇒ Set.empty[String],
-  eventAdapter:      EventAdapter[Event, _]                                      = NoOpEventAdapter.instance[Event],
-  snapshotWhen:      (State, Event, Long) ⇒ Boolean                              = ConstantFun.scalaAnyThreeToFalse,
-  recovery:          Recovery                                                    = Recovery(),
-  onSnapshot:        (ActorContext[Command], SnapshotMetadata, Try[Done]) ⇒ Unit = PersistentBehaviorImpl.defaultOnSnapshot[Command] _
+  persistenceId:       String,
+  emptyState:          State,
+  commandHandler:      PersistentBehaviors.CommandHandler[Command, Event, State],
+  eventHandler:        PersistentBehaviors.EventHandler[State, Event],
+  journalPluginId:     Option[String]                                              = None,
+  snapshotPluginId:    Option[String]                                              = None,
+  recoveryCompleted:   (ActorContext[Command], State) ⇒ Unit                       = ConstantFun.scalaAnyTwoToUnit,
+  tagger:              Event ⇒ Set[String]                                         = (_: Event) ⇒ Set.empty[String],
+  eventAdapter:        EventAdapter[Event, _]                                      = NoOpEventAdapter.instance[Event],
+  snapshotWhen:        (State, Event, Long) ⇒ Boolean                              = ConstantFun.scalaAnyThreeToFalse,
+  recovery:            Recovery                                                    = Recovery(),
+  supervisionStrategy: SupervisorStrategy                                          = SupervisorStrategy.stop,
+  onSnapshot:          (ActorContext[Command], SnapshotMetadata, Try[Done]) ⇒ Unit = PersistentBehaviorImpl.defaultOnSnapshot[Command] _
```

Comment:
As a bunch of these are going to be the same for all entities of a specific type, and doesn't change over time, should we extract them to a separate class that can have a shared instance across persistent entities instead of every entity repeating the fields?

Reply:
Not necessarily in this pr though